src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/HostAddress.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.connectivity;
+
+import software.amazon.awssdk.services.greengrassv2data.model.ConnectivityInfo;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class HostAddress {
+    private static final Pattern IPV4_PAT = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)(?::(\\d+)){0,1}$");
+    private static final Pattern IPV6_DOUBL_COL_PAT = Pattern.compile(
+            "^\\[{0,1}([0-9a-f:]*)::([0-9a-f:]*)(?:\\]:(\\d+)){0,1}$");
+    private static final Pattern IPV6_PAT = Pattern.compile(
+            "^\\[{0,1}([0-9a-f]+):([0-9a-f]+):([0-9a-f]+):([0-9a-f]+):([0-9a-f]+):"
+                    + "([0-9a-f]+):([0-9a-f]+):([0-9a-f]+)(?:\\]:(\\d+)){0,1}$");
+
+    private final String hostAddressString;
+
+
+    private HostAddress(String hostAddressString) {
+        this.hostAddressString = hostAddressString;
+    }
+
+    public static HostAddress of(String hostAddress) {
+        // TODO: validate hostAddress?
```

Comment:
Probably or make it value object (aka immutable object that acts a a value like `String` or `Boolean`)